packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3835,6 +3835,9 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
       _flushHistoryUpdates(rearrangeOverlay: false);
       assert(entry.route._popCompleter.isCompleted);
     }
+    if (_history.length == 1) {
```

Comment:
This does not consider the case where there is localhistory in the route. where the pop will pop the localhistory but the route won't be popped